core/ecschema-metadata/src/SchemaFormatsProvider.ts

```diff
@@ -60,6 +60,15 @@ export class SchemaFormatsProvider implements FormatsProvider {
     this.onFormatsChanged.raiseEvent({ formatsChanged });
   }
 
+  /** When using a presentation unit from a KindOfQuantity, the label and description should come from the KindOfQuantity */
+  private replaceWithKindOfQuantityProps(format: SchemaItemFormatProps, kindOfQuantity: KindOfQuantity): SchemaItemFormatProps {
```

Comment:
I think we should be returning the underlying format props rather than the schema item props ... at minimum we should use the base props interface.  stripping out the unnecessary props seems like a good idea ... if we end up serializing the extra pops they will end up in places we don't expect/want